src/Api/AdminConsole/Public/Models/Request/AssociationWithPermissionsRequestModel.cs

```diff
@@ -1,17 +1,27 @@
-﻿using Bit.Core.Models.Data;
+﻿using Bit.Core.Exceptions;
+using Bit.Core.Models.Data;
 
 namespace Bit.Api.AdminConsole.Public.Models.Request;
 
 public class AssociationWithPermissionsRequestModel : AssociationWithPermissionsBaseModel
 {
-    public CollectionAccessSelection ToCollectionAccessSelection()
+    public CollectionAccessSelection ToCollectionAccessSelection(bool migratedToFlexibleCollections)
     {
-        return new CollectionAccessSelection
+        var collectionAccessSelection =  new CollectionAccessSelection
         {
             Id = Id.Value,
             ReadOnly = ReadOnly.Value,
-            HidePasswords = HidePasswords.GetValueOrDefault(),
-            Manage = Manage.GetValueOrDefault()
+            HidePasswords = HidePasswords.GetValueOrDefault()
         };
+
+        // Throws if the org has not migrated to use FC but has passed in a Manage value in the request
+        if (!migratedToFlexibleCollections && Manage.HasValue)
+        {
+            throw new BadRequestException(
+                "Your organization has not migrated to use Flexible Collections and cannot make use of the Manage property.");
```

Comment:
Suggested alternative: "Your organization must be using the latest collection enhancements to use the Manage property." This aligns slightly more with the error message wording I added in https://github.com/bitwarden/server/pull/3671 (although that's still in code review so feedback is welcome).

In particular, flexible collections is an internal name only. If you like you can just swap out "Flexible Collections" for "collection enhancements" and that would also work.

Reply:
Totally fine with getting on the same page with our messaging. Thank you!